application/models/meetinginvite.go

```diff
@@ -0,0 +1,69 @@
+package models
+
+import (
+	"time"
+
+	"github.com/gobuffalo/pop"
+	"github.com/gobuffalo/validate"
+	"github.com/gobuffalo/validate/validators"
+	"github.com/gofrs/uuid"
+
+	"github.com/silinternational/wecarry-api/domain"
+)
+
+// MeetingInvite is the model for storing meeting invites sent to prospective users, linked to a meeting/event
+type MeetingInvite struct {
+	ID        int       `json:"id" db:"id"`
+	CreatedAt time.Time `json:"created_at" db:"created_at"`
+	UpdatedAt time.Time `json:"updated_at" db:"updated_at"`
+	MeetingID int       `json:"meeting_id" db:"meeting_id"`
+	InviterID int       `json:"inviter_id" db:"inviter_id"`
+	Secret    uuid.UUID `json:"secret" db:"secret"`
+	Email     string    `json:"email" db:"email"`
+}
+
+// MeetingInvites is used for methods that operate on lists of objects
+type MeetingInvites []MeetingInvite
+
+// Validate gets run every time you call one of: pop.ValidateAndSave, pop.ValidateAndCreate, pop.ValidateAndUpdate
+func (m *MeetingInvite) Validate(tx *pop.Connection) (*validate.Errors, error) {
+	return validate.Validate(
+		&validators.IntIsPresent{Field: m.MeetingID, Name: "MeetingID"},
+		&validators.IntIsPresent{Field: m.InviterID, Name: "InviterID"},
+		&validators.UUIDIsPresent{Field: m.Secret, Name: "Secret"},
+		&validators.EmailIsPresent{Field: m.Email, Name: "Email"},
+	), nil
+}
+
+// Create validates and stores the MeetingInvite data as a new record in the database.
+func (m *MeetingInvite) Create() error {
+	invite := *m
+	invite.Secret = domain.GetUUID()
+
+	err := create(&invite)
+	if err != nil && err.Error() ==
+		`pq: duplicate key value violates unique constraint "meeting_invites_meeting_id_email_idx"` {
+		err = nil
```

Comment:
In this case, is it important to hydrate `invite` with all of the existing MeetingInvite's values?

Reply:
Since it would involve a new database call, I think it's safe to wait to see if it's needed.